coderd/database/dbtestutil/db.go

```diff
@@ -88,6 +95,9 @@ func NewDB(t testing.TB, opts ...Option) (database.Store, pubsub.Pubsub) {
 		t.Cleanup(func() {
 			_ = sqlDB.Close()
 		})
+		if o.returnSQLDB != nil {
+			o.returnSQLDB(sqlDB)
+		}
```

Comment:
What happens if `dbtestutil.WillUsePostgres` is false? I could see this being a potential sharp edge. It would probably be a good idea to fail the test if this is defined and we're using the fakeDB.

Reply:
Good catch 👍🏻, I'll return an error.